X16 Reference - Appendix F - 65C816 Processor.md

```diff
@@ -0,0 +1,2470 @@
+
+Since the Native Mode has a distinct BRK vector, you do not need to query the
+stack to dispatch a BRK vs IRQ interrupt. You can just handle each immediately.
+
+See the [Vectors](#vectors) section for the break vector.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### BRL
+
+**Branch Long**
+
+```text
+SYNTAX         MODE          HEX LEN CYCLES      FLAGS   
+BRL LABEL      rel16         82  3   4           ........ .
+```
+
+BRL is a 16 bit _branch_ instruction, meaning assembly creates a relative
+address. Unlike BRA and the other branch instructions, BRL uses a 16-bit
+address, which allows for an offset of -32768 to 32767 bytes away from the
+instruction _following_ The BRL.
+
+Of course, due to wrapping of the 64K bank, this means that the entire 64K
+region is accessible. Values below 0 will simply wrap around and start from
+$FFFF, and values above $FFFF will wrap around to 0.
+
+Since this is a _relatve_ branch, that means code assembled with BRL, instead of
+JMP, can be moved around in memory without the need for re-assembly.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### BVC
+
+**Branch on Overflow Clear**
+
+```text
+SYNTAX         MODE          HEX LEN CYCLES      FLAGS   
+BVC LABEL      rel8          50  2   2+t+t*e*p   ........ .
+```
+
+Branches to the specified address when the Overflow bit is 0.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### BVS
+
+**Branch on Overflow Set**
+
+```text
+SYNTAX         MODE          HEX LEN CYCLES      FLAGS   
+BVS LABEL      rel8          70  2   2+t+t*e*p   ........ .
+```
+
+Branches to the specified address when the Overflow bit is 0.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### CLC
+
+**Clear Carry**
+
+```text
+SYNTAX         MODE          HEX LEN CYCLES      FLAGS   
+CLC            imp           18  1   2           .......c .
+```
+
+Clears the Carry bit in the flags. You'll usually use CLC before addition and
+SEC before subtraction. You'll also want to use CLC or SEC appropriately before
+calling certain KERNAL routines that use the **c** bit as an input value.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### CLD
+
+**Clear Decimal**
+
+```text
+SYNTAX         MODE          HEX LEN CYCLES      FLAGS   
+CLD            imp           D8  1   2           ....d... .
+```
+
+Clears the Decimal flag, returning the CPU to 8-bit or 16-bit binary operation.
+
+When Decimal is set, the CPU will store numbers in Binary Coded Decimal format.
+Clearing this flag restores the CPU to binary \(base 16\) operation. See
```

Comment:
True, but it's easier to think of hex vs BCD, rather than "binary" vs BCD.  I'm not the only person that's referred to it this way, including two of my college professors.

Reply:
The statement is confusing to me, because binary is base-2, not base-16.